homeworks/week2/JimLe-Homework.java

```diff
@@ -0,0 +1,32 @@
+class Main {
+    public static void main(String[] args) {
+        removeDuplicateCharacters("Hello world!".toCharArray());
+        removeDuplicateCharacters("Hi!".toCharArray());
+        removeDuplicateCharacters("H".toCharArray());
+        removeDuplicateCharacters("aaaaaaaaaaaaaaaaa".toCharArray());
+        removeDuplicateCharacters("test  case  ".toCharArray());
+        removeDuplicateCharacters("This is a sentense to test the program".toCharArray());
+        removeDuplicateCharacters("".toCharArray());
+        removeDuplicateCharacters(null);
+    }
+
+    // Time complexity is O(n)
+    // Space complexity is O(n) as we use an extra space for StringBuffer
```

Comment:
nit: O(N) is correct for average case and worst case.

But if you can talk about best case scenario it, it would be nicer. In the best case scenario "aaaaaaaaaaaaaaaaa", the space complexity is O(1).